capstone/src/main/webapp/js/maps.js

```diff
@@ -0,0 +1,68 @@
+/* eslint no-unused-vars: ["error", {"varsIgnorePattern": "createMap"}] */
+
+/**
+ * Function that creates map markers and info windows.
+ */
+function createMap() {
+  const map = new google.maps.Map(
+      document.getElementById('map'),
+      {center: {lat: 39.8097343, lng: -98.5556199}, zoom: 5});
+
+  const mapsDataList = fetchData();
+
+  const infoWindow = new google.maps.InfoWindow();
+  for (const mapsData of mapsDataList) {
+    const latitudeLongitude = new google.maps.LatLng(
+        mapsData['latitude'],
+        mapsData['longitude']);
+
+    const marker = new google.maps.Marker({
+      position: latitudeLongitude,
+      map: map,
+      title: mapsData['name'],
+    });
+
+    (function(marker, mapsData) {
+      google.maps.event.addListener(marker, 'click',
+          function() {
+            infoWindow.setContent(setDescription(
+                mapsData['name'],
+                mapsData['city'],
+                mapsData['isMainCampus']));
+            infoWindow.open(map, marker);
+          });
+    })(marker, mapsData);
+  }
+}
+
+/**
+ * Helper function to fetch data from servlet as JSON.
+ *
+ * @return {JSON} an array of dictionaries corresponding to relevant map data
+ */
+async function fetchData() {
+  // TODO(biancamacias): insert path to maps servlet
+  await fetch('/map')
+      .then((response) => response.json())
+      .then((mapsDataFromJSON) => {
+        console.log('Fetching data...');
+        console.log(mapsDataFromJSON);
+        return mapsDataFromJSON;
+      });
```

Comment:
```suggestion
      });
  return mapsDataFromJSON;
```

Reply:
i had this code before just like this, but i think javascript scope hinders me from doing this as i got an error saying it couldn't find `mapsDataFromJson` when i had the `return` outside the brackets. i haven't tested to see if it works with the servlet. i might end up just returning `response.json()` in the end for brevity, but im going to wait to address this.

Reply:
Oh yea since `mapsDataFromJSON` is only defined within the last `then` block, it won't be accessible outside of it. Then yea it can be left alone.